logstash-core/src/test/java/org/logstash/log/LogstashConfigurationFactoryTest.java

```diff
@@ -0,0 +1,83 @@
+package org.logstash.log;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.ThreadContext;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.routing.RoutingAppender;
+import org.apache.logging.log4j.core.config.Configuration;
+
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.test.appender.ListAppender;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static junit.framework.TestCase.assertNull;
+import static junit.framework.TestCase.assertEquals;
+
+public class LogstashConfigurationFactoryTest {
+
+    private static final String CONFIG = "log4j2-log-pipeline-test.properties";
+
+    @Before
+    public void setUp() {
+        System.setProperty("log4j.configurationFile", CONFIG);
+        System.setProperty("ls.log.format", "plain");
+        System.setProperty("ls.logs", "build/logs");
+        System.setProperty(LogstashConfigurationFactory.PIPELINE_SEPARATE_LOGS, "true");
+
+        ThreadContext.clearAll();
```

Comment:
Do you mean to to save the System.properties used + the ThreadContext in the @Before and restore it in the @After? To me sounds good, at the end the unit test results as side effect free to the System.properties

Reply:
Yes, something like that to avoid the test having any unforeseen side effects - we might even be able to use `@BeforeClass` and `@AfterClass`
 to set and unset the properties, and an `@Rule` to handle the ThreadContext - 

```
 @Rule
    public ThreadContextRule threadContextRule = new ThreadContextRule();
```

Reply:
Yes I do it, I didn't used the `@Rule` because in JUnit5 does not exists rules as I read, so be open moving to JUnit5 without incur in deprecated stuff.

Reply:
Ah! I missed that `@Rule` was a junit5 thing. No worries!

Reply:
fixed by 3ab931e95920da9944f78fc442c26dde9e71ed74